src/lib.rs

```diff
@@ -2151,6 +2170,12 @@ impl Build {
         };
 
         let (sdk_prefix, sim_prefix, min_version) = match os {
+            Os::MacOs => (
+                "macosx",
+                "",
+                std::env::var("MACOSX_DEPLOYMENT_TARGET")
+                    .unwrap_or_else(|_| (if self.cpp { "10.9" } else { "10.4" }).into()),
```

Comment:
As I mentioned in the other PR, I'd prefer this match the logic in https://github.com/rust-lang/rust/blob/6365e5ad9fa9e2ec867a67aeeae414e7c62d8354/compiler/rustc_target/src/spec/apple_base.rs#L105-L114